draft-ietf-quic-qpack.md

```diff
@@ -871,21 +871,38 @@ the decoder.
 
 ### Blocked Eviction
 
-The encoder MUST NOT permit an entry to be evicted while a reference to that
-entry remains unacknowledged.  If a new header to be inserted into the dynamic
-table would cause the eviction of such an entry, the encoder MUST NOT emit the
-insert instruction until the reference has been processed by the decoder and
-acknowledged.
-
-The encoder can emit a literal representation for the new header in order to
-avoid encoding delays, and MAY insert the header into the table later if
-desired.
-
-To ensure that the blocked eviction case is rare, references to the oldest
-entries in the dynamic table SHOULD be avoided.  When one of the oldest entries
-in the table is still actively used for references, the encoder SHOULD emit an
-Duplicate representation instead (see {{duplicate}}).
+The encoder MUST NOT emit an instruction that evicts an entry while a reference
+to that entry remains unacknowledged.  When the encoder encounters a blocked
+eviction, it emits a literal representation for the new header, and MAY insert
+the header into the table later if desired.
+
+To ensure that blocked evictions are rare, the encoder SHOULD avoid referencing
+older entries that will be evicted soon.  Rather than reference such an entry,
+the encoder SHOULD emit a Duplicate instruction (see {{duplicate}}), and
+reference the duplicate instead.
+
+Determining which entries are too close to eviction to reference is an encoder
+preference.  One heuristic is to target a fixed amount of available space in the
+dynamic table: either unused space or space that can be reclaimed by evicting
+unreferenced entries.  To achieve this, the encoder can maintain a draining
+index, which is the smallest absolute index in the dynamic table that it will
+emit a reference for.  As new entries are inserted, the encoder increments the
+draining index such that the amount of unused space and space occupied by
+draining entries table is larger than its target threshold.  Draining entries
+will eventually become evictable as any references to them are acknowledged.
```

Comment:
"As new entries are inserted, the encoder increases the draining index to maintain the section of the table that it will not reference.  Draining entries - entries with an absolute index lower than this the draining index - will not accumulate new references.  The number of unacknowledged references to draining entries will eventually become zero, making the entry available for eviction."